### ISSUE TITLE
Tool reports a false positive on javax.mail-1.6.2.jar

### ISSUE THREAD

**Issue by swetan27** [anchore/anchore-engine#1105]
[

**Is this a request for help?**:

**Is this a BUG REPORT or a FEATURE REQUEST?** (choose one):
BUG REPORT

**Version of Anchore Engine and Anchore CLI if applicable**: anchore-cli, version 0.8.2
Name: anchore-engine
Version: 0.8.2

**What happened**:
Tool reported a CVE-2017-15806 on com.sun.mail.javax.mail-1.6.2.jar

False detection in the com.sun.mail.javax.mail-1.6.2.jar. It is adding CVE - https://nvd.nist.gov/vuln/detail/CVE-2017-15806. Inspite of manifest and vuln.json capturing the cpe as below, the analyzer reports a vulnerability related to Zeta Components Mail
"package_cpe": "cpe:a:sun:mail:1.6.2:*:*",
"package_cpe23": "cpe:2.3:a:sun:mail:1.6.2:*:-:-:-:-:-:*", 

As per NVD, the affected component is:  cpe:2.3:a:zetacomponents:mail:*:*:*:*:*:*:*:*

Snippet from manifest file:
![image](https://user-images.githubusercontent.com/45357711/121186754-b479cd00-c867-11eb-95e1-6804b092a109.png)

Snippet from Anchor's vuln report.json:
![image](https://user-images.githubusercontent.com/45357711/121186926-d410f580-c867-11eb-9a15-92d7313cef58.png)

**What did you expect to happen**:
The tool analyser should not have matched the package cpe with the vulnerable, and since no match, should have passed the scan.

**Any relevant log output from /var/log/anchore**:
Shared the snippet, as no error in Anchore working

**What docker images are you using**: 
openjdk:8u282-jre-slim-buster

**How to reproduce the issue**:
Create and scan simple Java project using which is using latest com.sun.mail.javax.mail-1.6.2.jar

**Anything else we need to know**:
No

**Comment by swetan27** [comment]
@zhill - Kindly look into the issue

**Comment by zburstein** [comment]
@swetan27 I see that you are using Version: 0.8.2 of engine. Since that version we have introduced a number of changes/enhancements to how engine analyzes images. As of v0.9.0 we use [syft](https://github.com/anchore/anchore-engine/issues/1106) to analyze images and we have introduced a number of patch releases that address issues specific to java packages.  Does running the image in question again the most recent version of engine (v0.10) produce the same false positive?
